hearpreprocess/pipeline.py

```diff
@@ -538,14 +538,6 @@ def split_k_folds(self, metadata: pd.DataFrame):
         """
         Deterministically split dataset into k-folds
         """
-
-        splits_present = metadata["split"].unique()
-        if len(splits_present) > 1:
-            raise AssertionError(
-                "More than one split found in dataset, there must be only one split "
-                "to create new k-folds from."
-            )
-
         # Deterministically sort all unique split_keys.
         split_keys = sorted(metadata["split_key"].unique())
 
```

Comment:
Yes since we are doing the `array_split` on all the `split_key` in the metadata, all the data points are considered.
One thing though is, maybe we can remove the split column for the `new_split_kfold` mode, i.e. assert that there is no column called `split` in the metadata. Like we make three columns -- relpath, label and split. So we can make sure there is no split for this mode.